libs/ngxtension/host-binding/src/host-binding.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import {
+	effect,
+	ElementRef,
+	HostBinding,
+	inject,
+	Injector,
+	Renderer2,
+	RendererStyleFlags2,
+	runInInjectionContext,
+	Signal,
+	WritableSignal,
+} from '@angular/core';
+import { assertInjector } from 'ngxtension/assert-injector';
+
+/**
+ * `hostBinding` takes a `hostPropertyName` to attach a data property, a class, a style or an attribute (as `@HostBinding` would) to the host.
+ * The udpate is applied based on the update of the provided signal (writable or not).
+ *
+ * @param {Required<HostBinding>['hostPropertyName']} hostPropertyName - the same property that is bound to a data property, a class, a style or an attribute as `@HostBinding`.
+ * @param {Signal | WritableSignal} signal - the signal on which to react to changes to update the host, and the one that will be returned as it is
+ * @returns {Signal | WritableSignal}
+ *
+ * @example
+ * ```ts
+ * export class MyComponent {
+ *  readonly background = hostBinding('style.background', signal('blue'));
+ *
+ *  constructor() {
+ *    setTimeout(() => this.background.set('red'), 3000);
+ *  }
+ * }
+ * ```
+ */
+export function hostBinding<T, S extends Signal<T> | WritableSignal<T>>(
+	hostPropertyName: Required<HostBinding>['hostPropertyName'],
+	signal: S,
+	injector?: Injector
+): S {
+	injector = assertInjector(hostBinding, injector);
+
+	runInInjectionContext(injector, () => {
+		const renderer = inject(Renderer2);
+		const element = inject(ElementRef).nativeElement;
+
+		effect(() => {
+			const value = signal();
+			const [binding, property, unit] = hostPropertyName.split('.');
+
+			(
+				({
+					style: () =>
+						renderer.setStyle(
+							element,
+							property,
+							`${value}${unit ?? ''}`,
+							property.startsWith('--')
+								? RendererStyleFlags2.DashCase
+								: undefined
+						),
+					attr: () => renderer.setAttribute(element, property, String(value)),
+					class: () => {
+						if (value) {
+							renderer.addClass(element, property);
+						} else {
+							renderer.removeClass(element, property);
+						}
+					},
+				})[binding] ?? (() => renderer.setProperty(element, binding, value))
+			)();
```

Comment:
nit: I think this is trying to be too smart here. Does this have to be recreated every time the effect triggers? I would rather create the object outside of the `effect` or just good ol' if/elseif instead.

Reply:
Well, you are right. I really like this syntax that has the exact same perfs as a switch/case (when not in a loop) but here it makes it less readable. Therefore I should stick with the old good switch/case (I wish that we will have the same one as java someday :stuck_out_tongue:)

About perfs it should not make such a big diff here, though. Even if it is within an effect (thus, a "loop"), variables with host binding linked to them are (usually ?) updated a few times. **But** we never know, so once again, I should refactor it to use a switch/case instead!